src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -680,12 +687,17 @@ MavlinkReceiver::handle_message_hil_optical_flow(mavlink_message_t *msg)
 	/* Use distance value for distance sensor topic */
 	distance_sensor_s d{};
 
+	device::Device::DeviceId device_id;
+	device_id.devid_s.bus = device::Device::DeviceBusType::DeviceBusType_SERIAL;
+	device_id.devid_s.devtype = DRV_DIST_DEVTYPE_MAVLINK;
```

Comment:
I think it's also an open question if mavlink should be a "bus type", but this works for now.

Reply:
Can easily add that if desired

Reply:
I think it makes sense then we'll know how to decode the device id with meaningful mavlink metadata. FYI @CarlOlsson

Reply:
Ok, just created a branch for that.  PR coming next